### PR TITLE
Add generated docs for ContractKit

### LINKED ISSUE #2498: Devs SBAT reference docs on entire contractkit generated by jsdoc annotations
### Expected Behavior

Extensive docs (at minimum) all kit contract wrappers

### Current Behavior

Few docs for kit (getting started and examples) manually populated

### PR DESCRIPTION
**Pull request by yorhodes**
### Description

Adds typedoc generation from contractkit TS project to `/docs/developer-resources/contractkit/reference` with `yarn docs` (uses `--theme gitbook` to generate markdown content). This generator expects the output directory to be the root of the gitbook project, but in our monorepo case, it is not. As a workaround, there are placeholder tags in the root `docs/SUMMARY.md`. `contractkit/linkdocs.ts` fills these tags to resolve "nested gitbook" projects. With a little work, `linkdocs` can be used to integrate generated docs for any monorepo TS package.

### Tested

`yarn --cwd packages/docs preview`

### Other changes

Adds `yarn preview` to docs packages

### Related issues

- Fixes #2498

### CODE REVIEW

**Review by timmoreton — APPROVED** [review]
LGTM.  Didn't see anything that checks contractkit docs remain up to date on every commit, like we have for CLI. Should we?

File: packages/contractkit/linkdocs.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import { readFileSync, writeFileSync } from 'fs'
```
Comment (jmrossy) [inline packages/contractkit/linkdocs.ts]: nit, move this to a `scripts` folder

**Review by jmrossy — COMMENTED** [review]
Looks good! Just one nit below

File: packages/docs/developer-resources/contractkit/reference/classes/_address_registry_.addressregistry.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+* [addressFor](_address_registry_.addressregistry.md#addressfor)
+* [allAddresses](_address_registry_.addressregistry.md#alladdresses)
+
+## Constructors
+
+###  constructor
+
+\+ **new AddressRegistry**(`kit`: [ContractKit](_kit_.contractkit.md)): *[AddressRegistry](_address_registry_.addressregistry.md)*
+
+*Defined in [packages/contractkit/src/address-registry.ts:18](https://github.com/celo-org/celo-monorepo/blob/06adf8b7a/packages/contractkit/src/address-registry.ts#L18)*
```
Comment (yorhodes) [inline packages/docs/developer-resources/contractkit/reference/classes/_address_registry_.addressregistry.md]: the blob hash changes every commit so the CI `git diff` check doesn't work exactly like the CLI one

Comment (yorhodes) [reply packages/docs/developer-resources/contractkit/reference/classes/_address_registry_.addressregistry.md]: @timmoreton

**Review by mcortesi — APPROVED** [review]
CI check for contractKit docs seems to be failing

**Pull request merged**